e2e/cypress/integration/team_settings/closed_team_invite_with_specific_domain_spec.js

```diff
@@ -0,0 +1,141 @@
+
+// Group: @team_settings
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+import {getEmailUrl, getEmailMessageSeparator, reUrl, getRandomId} from '../../utils';
+
+describe('Team Settings', () => {
+    let testTeam;
+    const randomId = getRandomId();
+    const username = `user${randomId}`;
+    const email = `user${randomId}@sample.mattermost.com`;
+    const password = 'passwd';
+    const emailDomain = 'sample.mattermost.com';
+
+    const baseUrl = Cypress.config('baseUrl');
+    const mailUrl = getEmailUrl(baseUrl);
+    let isLicensed;
+
+    before(() => {
+        // # If the instance the test is running on is licensed, assign true to isLicensed variable
+        cy.apiGetClientLicense().then(({license}) => {
+            isLicensed = license.IsLicensed === 'true';
+        });
+
+        // # Disable LDAP and do email test if setup properly
+        cy.apiUpdateConfig({LdapSettings: {Enable: false}});
+        cy.apiEmailTest();
+
+        cy.apiInitSetup().then(({team}) => {
+            testTeam = team;
+            cy.visit(`/${team.name}/channels/town-square`);
+        });
+    });
+
+    it('MM-T386 Invite new user to closed team with \'Allow only users with a specific email domain to join this team\' set to \'sample.mattermost.com\'', () => {
+        // # Open 'Team Settings' modal
+        cy.get('.sidebar-header-dropdown__icon').click();
+        cy.findByText('Team Settings').should('be.visible').click();
+
+        // * Check that the 'Team Settings' modal was opened
+        cy.get('#teamSettingsModal').should('exist').within(() => {
+            // * Ensure that 'Allow any user with an account on this server' is set to 'No'
+            cy.get('#open_inviteDesc').should('have.text', 'No');
+
+            // # Click on the 'Allow only users with a specific email domain to join this team' edit button
+            cy.get('#allowed_domainsEdit').should('be.visible').click();
+
+            // # Set 'sample.mattermost.com' as the only allowed email domain and save
+            cy.wait(TIMEOUTS.HALF_SEC);
+            cy.focused().type(emailDomain);
+            cy.findByText('Save').should('be.visible').click();
+
+            // # Close the modal
+            cy.get('#teamSettingsModalLabel').find('button').should('be.visible').click();
+        });
+
+        // # Open the 'Invite People' full screen modal
+        cy.get('.sidebar-header-dropdown__icon').click();
+        cy.get('#invitePeople').find('button').eq(0).click();
+
+        // # Wait half a second to ensure that the modal has been fully loaded
+        cy.wait(TIMEOUTS.HALF_SEC);
+
+        if (isLicensed) {
+            // # Click "Invite members"
+            cy.findByTestId('inviteMembersLink').should('be.visible').click();
+        }
+
+        cy.findByRole('textbox', {name: 'Add or Invite People'}).type(email, {force: true}).wait(TIMEOUTS.HALF_SEC).type('{enter}', {force: true});
+        cy.get('#inviteMembersButton').click();
```

Comment:
I've seen this pattern in your other PRs and I thought it would be better to extract it out into a function, located in a shared `helpers.js` within `/team_settings` folder.